tests/e2e/specs/editor/media/insertMediaFromDialog.js

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { percySnapshot } from '@percy/puppeteer';
+
+/**
+ * Internal dependencies
+ */
+import { createNewStory } from '../../../utils';
+
+const MODAL = '.media-modal';
+
+describe('Inserting Media from Dialog', () => {
+  // Uses the existence of the element's frame element as an indicator for successful insertion.
+  it('should insert an image by clicking on it', async () => {
+    await createNewStory();
+
+    await expect(page).not.toMatchElement('[data-testid="FrameElement"]');
+
+    // Clicking will only act on the first element.
+    await expect(page).toClick('[data-testid="uploadButton"]');
+
+    await page.waitForSelector(MODAL, {
+      visible: true,
+    });
+    const btnTab = '#menu-item-browse';
+    await page.waitForSelector(btnTab);
+    await page.evaluate((selector) => {
+      document.querySelector(selector).click();
+    }, btnTab);
+    const btnSelector =
+      '.attachments-browser .attachments .attachment:first-of-type';
+    await page.waitForSelector(btnSelector);
+    await page.evaluate((selector) => {
+      document.querySelector(selector).click();
+    }, btnSelector);
+    const btnSelect = '.media-button-select';
+    await page.evaluate((selector) => {
+      document.querySelector(selector).click();
+    }, btnSelect);
+
+    await expect(page).toMatchElement('[data-testid="imageElement"]');
```

Comment:
Can we use the existing `data-testid="frameElement"`? That works for all elements and not just images

Reply:
Yeah, but the point here is to make sure that the element is an image. I don't want to a selector that works on all elements.

Reply:
Then I'd rather define this test ID in `assets/src/edit-story/components/canvas/frameElement.js`, not in the image's display component.

e.g. changing `data-testid="frameElement"` to 
```
data-testid={`frameElement-${element.type}`}
```

Just need to update `assets/src/edit-story/elements/image/karma/resourceLoading.karma.js` accordingly to use `fixture.screen.getAllByTestId('frameElement-image')`

Reply:
Changing this is a bigger change than that, frameElement is referenced in karma and e2e test.